lib/bitstyles_phoenix/component/heading.ex

```diff
@@ -0,0 +1,280 @@
+defmodule BitstylesPhoenix.Component.Heading do
+  use BitstylesPhoenix.Component
+
+  @moduledoc """
+  Components for building headings.
+  """
+
+  @doc ~s"""
+  Render a page header with a `<h1>` title and an optional `title_extra` and some `action` slots
+  for that page (usually combined with buttons and dropdowns).
+
+  ## Attributes
+  - `class` - Set CSS classes on the outer div.
+  - All other attributes are passed to the outer `div` tag.
+
+  ## Attributes - `title_extra` slot
+  - `class` - Set CSS classes on the surrounding div.
+  - All other attributes are passed to the outer `div` tag.
+
+  ## Attributes - `action` slots
+  - `class` - Set CSS classes on the surrounding `li`.
+  - All other attributes are passed to the outer `li` tag.
+
+  See [the page header docs](https://bitcrowd.github.io/bitstyles/?path=/docs/ui-content-page-header--page-header)
+  for more examples and details.
+  """
+
+  story(
+    "Page title",
+    '''
+        iex> assigns = %{}
+        ...> render ~H"""
+        ...> <.ui_page_title>
+        ...>   Title
+        ...> </.ui_page_title>
+        ...> """
+        """
+        <div class="u-flex u-justify-between u-flex-wrap u-items-center">
+          <div class="u-flex u-flex-wrap u-items-center">
+            <h1 class="u-margin-m-right u-break-text">
```

Comment:
why margin right? wouldn't it make more sense to have a margin-left on the optional title extra?

Reply:
Maybe 🤷  It's copied from darrens examples though. Maybe it also helps with the mobile screen, when the action buttons collapse onto it?

Reply:
yes, that is the reason